app/controllers/master_files_controller.rb

```diff
@@ -45,9 +47,8 @@ def show
 
   def embed
     @master_file = MasterFile.find(params[:id])
-    if can? :read, @master_file
-      @token = @master_file.nil? ? "" : StreamToken.find_or_create_session_token(session, @master_file.id)
-      @stream_info = @master_file.stream_details(@token, default_url_options[:host])
+    if can? :read, @master_file.media_object
```

Comment:
Why are you digging into media_object here and not letting cancan determine this itself?